js/gateio.js

```diff
@@ -637,7 +637,7 @@ module.exports = class gateio extends Exchange {
         }
         const resultLength = result.length;
         if (resultLength === 0) {
-            throw new ExchangeError (this.id + " does not support '" + type + "' type, set exchange.options['defaultType'] to " + "'spot', 'margin', 'swap', 'future' or 'option'"); // eslint-disable-line quotes
+            throw new ExchangeError (this.id + " does not support '" + type + "' type, set exchange.options['defaultType'] to " + "'spot', 'swap', 'future' or 'option'"); // eslint-disable-line quotes
```

Comment:
Is there a reason for removing the `'margin'` type here? We still support it on line 629 and still have to support it in private methods.

Reply:
We're supporting it for backwards compatability but I thought we were getting rid of it? MarginType of 'cross' and 'isolated' covers this

Reply:
> but I thought we were getting rid of it?

We have to keep `'margin'` as one of the values for the defaultType option. Say, how do you cancel all open margin orders with cancelAllOrders later (suppose we will have to add that), for example? We will still need the defaultType = margin and the marginType = cross / isolated. So, removing the `'margin'` type from that exception message would be somewhat misleading, cause it is supported. On a sidenote, marginType = 'spot' is semantically wrong.

Reply:
> On a sidenote, marginType = 'spot' is semantically wrong. The goal is to avoid mixing the defaultType with the marginType.

The user would just not include a marginType, the only reason we set `marginType='spot'` is because we set the paramete `account` to the value of `marginType`, we could change `getMarginType` to `getAccountType` if you like

Reply:
> The user would just not include a marginType, the only reason we set marginType='spot' is because we set the paramete account to the value of marginType, we could change getMarginType to getAccountType if you like

Yes, that would make more sense.